### DISCUSSION TITLE
长轮询机制中客户端超时时间为啥比服务端要长？

### DISCUSSION THREAD

**Discussion by wangyuDH** [apolloconfig/apollo#3934] [Q&A] [answered]
//90 seconds, should be longer than server side's long polling timeout, which is now 60 seconds
  private static final int LONG_POLLING_READ_TIMEOUT = 90 * 1000;
请问下RemoteConfigLongPollService.doLongPollingRefresh  中设置的  客户端超时时间为啥比服务端要长?

**Comment by nobodyiam** [comment]
如果客户端比服务端短的话，那服务端还没返回，客户端就超时了

**Comment by wangyuDH** [comment]
我的理解：就算客户端超时了也没啥问题啊，超时报错还是会重新请求服务端啊，因为源码里面是while(true)的死循环啊。
while (!m_longPollingStopped.get() && !Thread.currentThread().isInterrupted()) {
   XXX;
   final HttpResponse<List<ApolloConfigNotification>> response =
            m_httpUtil.doGet(request, m_responseType);
  XXX;
}

**Comment by nobodyiam** [comment, marked as answer]
抽象解释一下吧，客户端 A 调用服务端 B 的服务，如果 B 的处理时间是 5 秒，那么 A 的超时时间设置就要大于 5 秒，如果小于 5 秒，那么调用就都失败了。
